File: reactive-demo-backend/src/main/java/dev/divezone/demo/reactive/web/AirlineController.java
package dev.divezone.demo.reactive.web;

import static org.springframework.web.bind.annotation.RequestMethod.GET;

import java.util.List;

import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import dev.divezone.demo.reactive.data.AirlineRepository;
import dev.divezone.demo.reactive.model.Airline;
import lombok.AllArgsConstructor;

@RestController
@RequestMapping(value = "/api/airline", produces = "application/json")
@AllArgsConstructor
public class AirlineController {

    private final AirlineRepository repository;

    @RequestMapping(value = "/{country}", method = GET)
    public List<Airline> getForCountry(
            @PathVariable final String country
    ) {
        return repository.findAirlineByCountry(country);
    }

    @RequestMapping(value = "/{id}/{favorite}", method = GET)
    public Airline setFavorite(
            @PathVariable final Integer id,
            @PathVariable final boolean favorite
    ) {
        var result = repository.findById(id);
        if (result.isPresent()) {
            var airline = result.get();
            airline.setFavorite(favorite);
            repository.save(airline);
            return airline;
        }
        return null;
    }

    @RequestMapping(value = "/", method = GET)
    public List<Airline> index() {
        return repository.findAirlineByCountry("Netherlands");
    }

}

File: reactive-demo-backend-kotlin/src/main/kotlin/dev/divezone/demo/reactive/web/AirlineController.kt
package dev.divezone.demo.reactive.web

import dev.divezone.demo.reactive.data.AirlineRepository
import dev.divezone.demo.reactive.model.Airline
import org.springframework.data.repository.findByIdOrNull
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod.GET
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(value = ["/api/airline"], produces = ["application/json"])
class AirlineController(val repository: AirlineRepository) {

    @RequestMapping(value = ["/{country}"], method = [GET])
    fun getForCountry(
            @PathVariable country: String
    ) = repository.findAirlineByCountry(country)

    @RequestMapping(value = ["/{id}/{favorite}"], method = [GET])
    fun setFavorite(
            @PathVariable id: Int,
            @PathVariable favorite: Boolean
    ): Airline? = repository.findByIdOrNull(id)?.let {
        it.favorite = favorite
        repository.save(it)
        it
    }

    @RequestMapping(method = [GET])
    fun index() = repository.findAirlineByCountry("Netherlands")
}